### PR TITLE
`Development`: Adjust server tests to restructuring into modules

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
After restructuring the server (prod) module structure, these changes adjust the test-code structure to the new prod-code structure.

**Note**: Most of the changes just come from moving files and methods.

### Description
In detail, this includes:
- Moving tests to the same modules in test code that exist in prod code incl. adjusting (file) paths
- Adding `test_repository` subpackage to each module if this module uses a test repository
- Each existing TestRepository implements the respective (prod) Repository. If such a test repository exists, the prod repository must not be used in any test (-> arch test)
- Methods in prod repositories that are only used by test code must reside in the respective test repository (arch test)
- Each TestRepository must be annotated with @Primary as otherwise, this causes conflicts with the prod-repository (arch test)
- `AnswerMessageIntegrationTest#testCreateAnswerInExamChannel` has been disabled. This test fails with `JsonException No _valueDeserializer assigned` but only if run as part of the whole test suite. The root cause for this issue is unclear.

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

## Summary by CodeRabbit

- **Bug Fixes**
	- Removed several methods from various repository interfaces, which may affect how users retrieve specific data.
	- Adjusted the functionality for retrieving user and course-related data, streamlining the user experience.

- **Chores**
	- Reorganized the `AuditResource` class into a more specific package for better project structure.

### CODE REVIEW

**Review by cremertim — DISMISSED** [review]
Code changes look good to me. Mostly package imports changed to the new structure

**Review by SimonEntholzer — DISMISSED** [review]
huge refactor :+1: all the deleted queries were not used?

**Comment by MaximilianAnzinger** [comment]
@SimonEntholzer 
First of all not many queries where deleted, we simply moved them out of the production code into test repositories, if they're not used in production code. However, the ones we deleted were indeed unused.

**Review by SimonEntholzer — DISMISSED** [review]

**Review by cremertim — DISMISSED** [review]
Re-approve after new commit

File: src/test/java/de/tum/cit/aet/artemis/communication/AnswerMessageIntegrationTest.java [hunk 1]
```diff
@@ -155,6 +156,7 @@ void testCreateAnswerInExerciseChannel() throws Exception {
     }
 
     @Test
+    @Disabled // ToDo: Find out why "JsonException No _valueDeserializer assigned" is thrown when running the whole test suite
```
Comment (raffifasaro) [inline src/test/java/de/tum/cit/aet/artemis/communication/AnswerMessageIntegrationTest.java:159]: Is this meant for later fixes and part of the PR?

**Review by raffifasaro — CHANGES_REQUESTED** [review]

File: src/test/java/de/tum/cit/aet/artemis/communication/AnswerMessageIntegrationTest.java (hunk 1 above)
Comment (ole-ve) [reply src/test/java/de/tum/cit/aet/artemis/communication/AnswerMessageIntegrationTest.java:159]: Not part of this PR do I get it finally merged and produce less merge conflicts.

**Review by raffifasaro — APPROVED** [review]
code LGTM

**Review by SimonEntholzer — APPROVED** [review]

**Pull request merged**